src/coverlet.core/Coverage.cs

```diff
@@ -420,11 +420,15 @@ private void CalculateCoverage()
                         var hitLocation = result.HitCandidates[i];
                         var document = documentsList[hitLocation.docIndex];
                         int hits = br.ReadInt32();
+                        hits = hits < 0 ? int.MaxValue : hits;
 
                         if (hitLocation.isBranch)
                         {
                             var branch = document.Branches[new BranchKey(hitLocation.start, hitLocation.end)];
                             branch.Hits += hits;
+
+                            if (branch.Hits < 0)
```

Comment:
There's a tiny risk that we add `int.MaxValue` to 1 resulting in 0, but that can be handled here by checking for any non-positive sum (assuming we've already checked that `hits != 0`, as suggested above):

```suggestion
                            if (branch.Hits <= 0)
```